File: apps/customer_mobile/lib/core/utils/business_kind.dart
/// Restaurant vs marketplace store (matches backend BusinessKind).
bool isRestaurantKind({String? kind, String? typeSlug}) {
  if (kind == 'RESTAURANT') return true;
  return typeSlug == 'restaurant';
}

bool isStoreKind({String? kind, String? typeSlug}) {
  if (kind == 'RESTAURANT') return false;
  if (kind == 'STORE') return true;
  return typeSlug != null && typeSlug != 'restaurant';
}

List<T> filterStoreBusinesses<T>(
  List<T> items, {
  required String? Function(T) kindOf,
  required String? Function(T) typeSlugOf,
}) {
  return items
      .where(
        (b) => isStoreKind(kind: kindOf(b), typeSlug: typeSlugOf(b)),
      )
      .toList();
}

File: frontend/src/lib/business-kind.ts
export type BusinessKind = 'RESTAURANT' | 'STORE';

export function isRestaurantKind(
  kind?: string | null,
  typeSlug?: string | null,
  businessTypeId?: string | null,
): boolean {
  if (typeSlug === 'restaurant') return true;
  if (businessTypeId == null && typeSlug == null) return true;
  return kind === 'RESTAURANT';
}

export function isStoreKind(
  kind?: string | null,
  typeSlug?: string | null,
  businessTypeId?: string | null,
): boolean {
  return !isRestaurantKind(kind, typeSlug, businessTypeId);
}

/** Marketplace merchants only (excludes restaurants for home banner / shops list). */
export function filterStoreBusinesses<T extends { kind?: string | null; businessType?: { slug?: string | null } | null }>(
  items: T[],
): T[] {
  return items.filter((b) => isStoreKind(b.kind, b.businessType?.slug));
}
